File: web/documentserver-example/python/src/configuration/configuration_tests.py
#
# (c) Copyright Ascensio System SIA 2023
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

from os import environ
from unittest import TestCase
from unittest.mock import patch
from urllib.parse import urlparse
from . import ConfigurationManager

class ConfigurationManagerTests(TestCase):
    def test_corresponds_the_latest_version(self):
        config_manager = ConfigurationManager()
        self.assertEqual(config_manager.version, '1.6.0')

class ConfigurationManagerExampleURLTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        url = config_manager.example_url()
        self.assertIsNone(url)

    @patch.dict(environ, {
        'EXAMPLE_URL': 'http://localhost'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        url = config_manager.example_url()
        self.assertEqual(url.geturl(), 'http://localhost')

class ConfigurationManagerDocumentServerPublicURLTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_public_url()
        self.assertEqual(url.geturl(), 'http://document-server')

    @patch.dict(environ, {
        'DOCUMENT_SERVER_PUBLIC_URL': 'http://localhost'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_public_url()
        self.assertEqual(url.geturl(), 'http://localhost')

class ConfigurationManagerDocumentServerPrivateURLTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_private_url()
        self.assertEqual(url.geturl(), 'http://document-server')

    @patch.dict(environ, {
        'DOCUMENT_SERVER_PRIVATE_URL': 'http://localhost'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_private_url()
        self.assertEqual(url.geturl(), 'http://localhost')

class ConfigurationManagerDocumentServerAPIURLTests(TestCase):
    @patch.object(
        ConfigurationManager,
        'document_server_public_url',
        return_value=urlparse('http://localhost')
    )
    def test_assigns_a_default_value(self, _):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_api_url()
        self.assertEqual(
            url.geturl(),
            'http://localhost/web-apps/apps/api/documents/api.js'
        )

    @patch.object(
        ConfigurationManager,
        'document_server_public_url',
        return_value=urlparse('http://localhost')
    )
    @patch.dict(environ, {
        'DOCUMENT_SERVER_API_PATH': '/api'
    })
    def test_assigns_a_value_from_the_environment(self, _):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_api_url()
        self.assertEqual(
            url.geturl(),
            'http://localhost/api'
        )

class ConfigurationManagerDocumentServerPreloaderURLTests(TestCase):
    @patch.object(
        ConfigurationManager,
        'document_server_public_url',
        return_value=urlparse('http://localhost')
    )
    def test_assigns_a_default_value(self, _):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_preloader_url()
        self.assertEqual(
            url.geturl(),
            'http://localhost/web-apps/apps/api/documents/cache-scripts.html'
        )

    @patch.object(
        ConfigurationManager,
        'document_server_public_url',
        return_value=urlparse('http://localhost')
    )
    @patch.dict(environ, {
        'DOCUMENT_SERVER_PRELOADER_PATH': '/preloader'
    })
    def test_assigns_a_value_from_the_environment(self, _):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_preloader_url()
        self.assertEqual(
            url.geturl(),
            'http://localhost/preloader'
        )

class ConfigurationManagerDocumentServerCommandURLTests(TestCase):
    @patch.object(
        ConfigurationManager,
        'document_server_private_url',
        return_value=urlparse('http://localhost')
    )
    def test_assigns_a_default_value(self, _):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_command_url()
        self.assertEqual(
            url.geturl(),
            'http://localhost/coauthoring/CommandService.ashx'
        )

    @patch.object(
        ConfigurationManager,
        'document_server_private_url',
        return_value=urlparse('http://localhost')
    )
    @patch.dict(environ, {
        'DOCUMENT_SERVER_COMMAND_PATH': '/command'
    })
    def test_assigns_a_value_from_the_environment(self, _):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_command_url()
        self.assertEqual(
            url.geturl(),
            'http://localhost/command'
        )

class ConfigurationManagerDocumentServerConverterURLTests(TestCase):
    @patch.object(
        ConfigurationManager,
        'document_server_private_url',
        return_value=urlparse('http://localhost')
    )
    def test_assigns_a_default_value(self, _):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_converter_url()
        self.assertEqual(
            url.geturl(),
            'http://localhost/ConvertService.ashx'
        )

    @patch.object(
        ConfigurationManager,
        'document_server_private_url',
        return_value=urlparse('http://localhost')
    )
    @patch.dict(environ, {
        'DOCUMENT_SERVER_CONVERTER_PATH': '/converter'
    })
    def test_assigns_a_value_from_the_environment(self, _):
        config_manager = ConfigurationManager()
        url = config_manager.document_server_converter_url()
        self.assertEqual(
            url.geturl(),
            'http://localhost/converter'
        )

class ConfigurationManagerJWTSecretTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        secret = config_manager.jwt_secret()
        self.assertEqual(secret, '')

    @patch.dict(environ, {
        'JWT_SECRET': 'your-256-bit-secret'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        secret = config_manager.jwt_secret()
        self.assertEqual(secret, 'your-256-bit-secret')

class ConfigurationManagerJWTHeaderTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        header = config_manager.jwt_header()
        self.assertEqual(header, 'Authorization')

    @patch.dict(environ, {
        'JWT_HEADER': 'Proxy-Authorization'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        header = config_manager.jwt_header()
        self.assertEqual(header, 'Proxy-Authorization')

class ConfigurationManagerJWTUseForRequest(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        use = config_manager.jwt_use_for_request()
        self.assertTrue(use)

    @patch.dict(environ, {
        'JWT_USE_FOR_REQUEST': 'false'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        use = config_manager.jwt_use_for_request()
        self.assertFalse(use)

class ConfigurationManagerSSLTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        enabled = config_manager.ssl_verify_peer_mode_enabled()
        self.assertFalse(enabled)

    @patch.dict(environ, {
        'SSL_VERIFY_PEER_MODE_ENABLED': 'true'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        enabled = config_manager.ssl_verify_peer_mode_enabled()
        self.assertTrue(enabled)

class ConfigurationManagerStoragePathTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        path = config_manager.storage_path()
        self.assertTrue(path.is_absolute())
        self.assertEqual(path.name, 'storage')

    @patch.dict(environ, {
        'STORAGE_PATH': 'directory'
    })
    def test_assigns_a_relative_path_from_the_environment(self):
        config_manager = ConfigurationManager()
        path = config_manager.storage_path()
        self.assertTrue(path.is_absolute())
        self.assertEqual(path.name, 'directory')

    @patch.dict(environ, {
        'STORAGE_PATH': '/directory'
    })
    def test_assigns_an_absolute_path_from_the_environment(self):
        config_manager = ConfigurationManager()
        path = config_manager.storage_path()
        self.assertEqual(f'{path}', '/directory')

class ConfigurationManagerMaximumFileSizeTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        size = config_manager.maximum_file_size()
        self.assertEqual(size, 5_242_880)

    @patch.dict(environ, {
        'MAXIMUM_FILE_SIZE': '10'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        size = config_manager.maximum_file_size()
        self.assertEqual(size, 10)

class ConfigurationManagerConversionTimeoutTests(TestCase):
    def test_assigns_a_default_value(self):
        config_manager = ConfigurationManager()
        timeout = config_manager.conversion_timeout()
        self.assertEqual(timeout, 120_000)

    @patch.dict(environ, {
        'CONVERSION_TIMEOUT': '10'
    })
    def test_assigns_a_value_from_the_environment(self):
        config_manager = ConfigurationManager()
        timeout = config_manager.conversion_timeout()
        self.assertEqual(timeout, 10)

File: web/documentserver-example/ruby/app/configuration/configuration_tests.rb
#
# (c) Copyright Ascensio System SIA 2025
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

# frozen_string_literal: true
# typed: true

require 'test/unit'
require_relative 'configuration'

# Enviroment module provides a mechanism for capturing and restoring the environment.
module Enviroment
  def initialize(name)
    @env = ENV.to_hash
    super
  end

  def setup
    ENV.replace(@env)
  end
end

# For testing the ConfigurationManager class.
class ConfigurationManagerTests < Test::Unit::TestCase
  def test_corresponds_the_latest_version
    config_manager = ConfigurationManager.new
    assert_equal(config_manager.version, '1.6.0')
  end
end

# For testing the example_uri method of ConfigurationManager.
class ConfigurationManagerExampleURITests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    uri = config_manager.example_uri
    assert_nil(uri)
  end

  def test_assigns_a_value_from_the_environment
    ENV['EXAMPLE_URL'] = 'http://localhost'
    config_manager = ConfigurationManager.new
    uri = config_manager.example_uri
    assert_equal(uri.to_s, 'http://localhost')
  end
end

# For testing the document_server_public_uri method of ConfigurationManager.
class ConfigurationManagerDocumentServerPublicURITests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_public_uri
    assert_equal(uri.to_s, 'http://documentserver')
  end

  def test_assigns_a_value_from_the_environment
    ENV['DOCUMENT_SERVER_PUBLIC_URL'] = 'http://localhost'
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_public_uri
    assert_equal(uri.to_s, 'http://localhost')
  end
end

# For testing the document_server_private_uri method of ConfigurationManager.
class ConfigurationManagerDocumentServerPrivateURITests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_private_uri
    assert_equal(uri.to_s, 'http://documentserver')
  end

  def test_assigns_a_value_from_the_environment
    ENV['DOCUMENT_SERVER_PRIVATE_URL'] = 'http://localhost'
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_private_uri
    assert_equal(uri.to_s, 'http://localhost')
  end
end

# For testing the document_server_api_uri method of ConfigurationManager.
class ConfigurationManagerDocumentServerAPIURITests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_api_uri
    assert_equal(
      uri.to_s,
      'http://documentserver/web-apps/apps/api/documents/api.js'
    )
  end

  def test_assigns_a_value_from_the_environment
    ENV['DOCUMENT_SERVER_API_PATH'] = '/api'
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_api_uri
    assert_equal(
      uri.to_s,
      'http://documentserver/api'
    )
  end
end

# For testing the document_server_preloader_uri method of ConfigurationManager.
class ConfigurationManagerDocumentServerPreloaderURITests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_preloader_uri
    assert_equal(
      uri.to_s,
      'http://documentserver/web-apps/apps/api/documents/preload.html'
    )
  end

  def test_assigns_a_value_from_the_environment
    ENV['DOCUMENT_SERVER_PRELOADER_PATH'] = '/preloader'
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_preloader_uri
    assert_equal(
      uri.to_s,
      'http://documentserver/preloader'
    )
  end
end

# For testing the document_server_command_uri method of ConfigurationManager.
class ConfigurationManagerDocumentServerCommandURITests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_command_uri
    assert_equal(
      uri.to_s,
      'http://documentserver/command'
    )
  end

  def test_assigns_a_value_from_the_environment
    ENV['DOCUMENT_SERVER_COMMAND_PATH'] = '/command'
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_command_uri
    assert_equal(
      uri.to_s,
      'http://documentserver/command'
    )
  end
end

# For testing the document_server_converter_uri method of ConfigurationManager.
class ConfigurationManagerDocumentServerConverterURITests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_converter_uri
    assert_equal(
      uri.to_s,
      'http://documentserver/converter'
    )
  end

  def test_assigns_a_value_from_the_environment
    ENV['DOCUMENT_SERVER_CONVERTER_PATH'] = '/converter'
    config_manager = ConfigurationManager.new
    uri = config_manager.document_server_converter_uri
    assert_equal(
      uri.to_s,
      'http://documentserver/converter'
    )
  end
end

# For testing the jwt_secret method of ConfigurationManager.
class ConfigurationManagerJWTSecretTests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    secret = config_manager.jwt_secret
    assert_equal(secret, '')
  end

  def test_assigns_a_value_from_the_environment
    ENV['JWT_SECRET'] = 'your-256-bit-secret'
    config_manager = ConfigurationManager.new
    secret = config_manager.jwt_secret
    assert_equal(secret, 'your-256-bit-secret')
  end
end

# For testing the jwt_header method of ConfigurationManager.
class ConfigurationManagerJWTHeaderTests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    header = config_manager.jwt_header
    assert_equal(header, 'Authorization')
  end

  def test_assigns_a_value_from_the_environment
    ENV['JWT_HEADER'] = 'Proxy-Authorization'
    config_manager = ConfigurationManager.new
    header = config_manager.jwt_header
    assert_equal(header, 'Proxy-Authorization')
  end
end

# For testing the jwt_use_for_request method of ConfigurationManager.
class ConfigurationManagerJWTUseForRequest < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    use = config_manager.jwt_use_for_request
    assert_true(use)
  end

  def test_assigns_a_value_from_the_environment
    ENV['JWT_USE_FOR_REQUEST'] = 'false'
    config_manager = ConfigurationManager.new
    use = config_manager.jwt_use_for_request
    assert_false(use)
  end
end

# For testing the ssl_verify_peer_mode_enabled method of ConfigurationManager.
class ConfigurationManagerSSLTests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    enabled = config_manager.ssl_verify_peer_mode_enabled
    assert_false(enabled)
  end

  def test_assigns_a_value_from_the_environment
    ENV['SSL_VERIFY_PEER_MODE_ENABLED'] = 'true'
    config_manager = ConfigurationManager.new
    enabled = config_manager.ssl_verify_peer_mode_enabled
    assert_true(enabled)
  end
end

# For testing the storage_path method of ConfigurationManager.
class ConfigurationManagerStoragePathTests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    path = config_manager.storage_path
    assert_true(path.absolute?)
    assert_equal(path.basename.to_s, 'storage')
  end

  def test_assigns_a_relative_path_from_the_environment
    ENV['STORAGE_PATH'] = 'directory'
    config_manager = ConfigurationManager.new
    path = config_manager.storage_path
    assert_true(path.absolute?)
    assert_equal(path.basename.to_s, 'directory')
  end

  def test_assigns_an_absolute_path_from_the_environment
    ENV['STORAGE_PATH'] = '/directory'
    config_manager = ConfigurationManager.new
    path = config_manager.storage_path
    assert_equal(path.to_s, '/directory')
  end
end

# For testing the maximum_file_size method of ConfigurationManager.
class ConfigurationManagerMaximumFileSizeTests < Test::Unit::TestCase
  include Enviroment

  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    size = config_manager.maximum_file_size
    assert_equal(size, 5_242_880)
  end

  def test_assigns_a_value_from_the_environment
    ENV['MAXIMUM_FILE_SIZE'] = '10'
    config_manager = ConfigurationManager.new
    size = config_manager.maximum_file_size
    assert_equal(size, 10)
  end
end

# For testing the convertation_timeout method of ConfigurationManager.
class ConfigurationManagerConversionTimeoutTests < Test::Unit::TestCase
  def test_assigns_a_default_value
    config_manager = ConfigurationManager.new
    timeout = config_manager.convertation_timeout
    assert_equal(timeout, 120)
  end
end
